src/app/ui/upload/upload.component.ts

```diff
@@ -84,7 +86,10 @@ export class UploadComponent implements OnInit {
       this.semester = parseInt(split[1].substr(5, 2), 10);
       const moedId = parseInt(split[3], 10);
       this.moed = (moedId === 1) ? 'A' : (moedId === 2) ? 'B' : 'C';
-      this.db.getCourse(courseId).subscribe(course => this.course = course);
+      this.db.getCourse(courseId).subscribe(course => {
+          this.course = course;
+          this.db.getFaculty(course.faculty).subscribe(faculty => this.faculty = faculty);
```

Comment:
It's not good. Every time there is new value you make a subscription to the inner observable. It results with memory leak.

Reply:
What to do here? 
This code is in the upload.component.ts file
![image](https://user-images.githubusercontent.com/19504086/51659209-9373f280-1fb2-11e9-97b2-d07f4faf9d7b.png)

Reply:
I already changed course to course$ which is an Observable<Course>

Reply:
maybe convert these observables to Promise? see if something like this works out:
course$.pipe(take(1)).toPromise().then(course => { do stuff});

Reply:
@ron4548 
tried this but the upload doesn't work :/
![image](https://user-images.githubusercontent.com/19504086/51668802-a5ae5a80-1fcb-11e9-8556-ea4676edd8cb.png)

Reply:
you need to return the promise returned by uploadScan